src/profile-logic/transforms.js

```diff
@@ -859,13 +916,96 @@ export function mergeFunction(
   });
 }
 
+/**
+ * Returns a Uint8Array filled with zeros and ones.
+ * result[funcIndex] === 1 iff isUnaccountedNativeFunction(funcIndex)
+ */
+function getUnaccountedNativeFunctions(thread: Thread): Uint8Array {
+  const { funcTable, stringTable } = thread;
+  const funcCount = funcTable.length;
+  const { isJS: funcIsJS, resource: funcResource, name: funcName } = funcTable;
+  const isUnaccountedNativeFunctionArr = new Uint8Array(funcTable.length);
+  for (let i = 0; i < funcCount; i++) {
+    if (funcIsJS[i] || funcResource[i] !== -1) {
+      continue;
+    }
+    const locationString = stringTable.getString(funcName[i]);
+    if (locationString.startsWith('0x')) {
```

Comment:
Thank you for asking me to justify this premature optimization :)

I was trying to avoid the cost from repeated lookups of the funcTable columns. But it turns out that calling `isUnaccountedNativeFunction` repeatedly is absolutely fast enough. On [the biggest profile I have](https://share.firefox.dev/3ATqEWT) (285092 functions), `getUnaccountedNativeFunctions` with repeated calls to `isUnaccountedNativeFunction` only takes 2ms, which is absolutely fast enough. The time spent in `getUnaccountedNativeFunctions` is dwarfed by the time it actually takes to apply the transform and create a new call node table - `computeCallNodeTable` takes 1500ms on that profile.